### DISCUSSION TITLE
Tmux does not persist after SSH session release

### DISCUSSION THREAD

**Discussion by jin0g** [ltsp/ltsp#235] [Q&A]
Hi,

I want to do session persistence via SSH with screen or tmux.
After the SSH session is released, the screen and tmux processes are terminated.
Need additional settings?

Thank you.

**Comment by alkisg** [comment]
Systemd now defaults to KillUserProcesses=yes, and LTSP enforces that, because many desktop environments do NOT properly kill the user processes on logout.
Debian-based distributions override the systemd defaults to KillUserProcesses=no, but this is bad; for example in Ubuntu or Debian MATE, even without LTSP, if one logs out and another user logs in, the processes of the previous user are still running, wasting memory and causing file-in-use issues.

So, the new, proper, systemd way of running processes outside the session is:

```shell
systemd-run --user --scope tmux
```

You might also need to enable lingering. See the following links for details:

https://unix.stackexchange.com/questions/5877/what-are-the-pros-cons-of-upstart-and-systemd/287282#287282

https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/ZNQW72UP36UAFMX53HPFFQTWTQDZVJ3M/

**Comment by jin0g** [comment]
@alkisg 
Thanks for the important information.
I will try to solve this with your information.
Thank you.
